File: scala/digital-wallet/src/main/scala/core/domain/entities/JournalEntry.scala
package core.domain.entities

import core.domain.enums.BalanceType.BalanceType
import core.domain.enums.SubwalletType.SubwalletType

case class JournalEntry(
  id: String,
  walletId: Option[String],
  subwalletType: SubwalletType,
  balanceType: BalanceType,
  amount: BigDecimal,
)

File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/core/domain/entities/JournalEntry.kt
package digitalwallet.core.domain.entities

import java.math.BigDecimal
import digitalwallet.core.domain.enums.SubwalletType
import digitalwallet.core.domain.enums.BalanceType

data class JournalEntry(
    val id: String,
    val walletId: String?,
    val subwalletType: SubwalletType,
    val balanceType: BalanceType,
    val amount: BigDecimal,
)